### PR TITLE
Fix Sarif Version parser error message

### LINKED ISSUE #45644: Error message for invalid errorlog format option is incorrect
**Version Used**: 

3.6.0-4.20251.5 (910223b6)

**Steps to Reproduce**:

1. Issue the command

    ```
    csc Program.cs /errorlog:Program.sarif,version=2.1.0
    ```

**Expected Behavior**:

The error message should be

```
error CS2046: Command-line syntax error: 'Program.sarif,version=2.1.0' is not a valid value for the '/errorlog:' option.
The value must be of the form '<file>[,version={1|1.0|2|2.1}]'.
```

**Actual Behavior**:

The error message is

```
error CS2046: Command-line syntax error: 'Program.sarif,version=2.1.0' is not a valid value for the '/errorlog:' option.
The value must be of the form '<file>[,version={1|1.0|1.0.0|2|2.1|2.1.0}]'.
```

... even though `1.0.0` and `2.1.0` are not accepted by the command line parser.

The same thing happens with VB.

### PR DESCRIPTION
**Pull request by eddynaka**
Fixes #45644

### CODE REVIEW

**Comment by eddynaka** [comment]
hi @jcouv / @AlekseyTs, i'm taking the lead on the following implementation that @lgolding started a few months ago: https://github.com/dotnet/roslyn/pull/45945

I added a few tests checking for 1.0.0 and 2.1.0 (the versions that I removed) for C#.
For VB, I don't have much knowledge, so I copied/pasted a test and changed the values.

Let me know what do you think.

Thank you.

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs [hunk 1]
```diff
@@ -4082,19 +4082,23 @@ public void ParseErrorLog()
...
 
             // Invalid SARIF version.
-            const string InvalidSarifVersion = @"C:\MyFolder\MyBinary.xml,version=42";
-            parsedArgs = DefaultParse(new[] { $"/errorlog:{InvalidSarifVersion}", "a.cs" }, baseDirectory);
-            parsedArgs.Errors.Verify(
-                // error CS2046: Command-line syntax error: 'C:\MyFolder\MyBinary.xml,version=42' is not a valid value for the '/errorlog:' option. The value must be of the form '<file>[,version={1|1.0|1.0.0|2|2.1|2.1.0}]'.
-                Diagnostic(ErrorCode.ERR_BadSwitchValue).WithArguments(InvalidSarifVersion, "/errorlog:", CSharpCommandLineParser.ErrorLogOptionFormat));
-            Assert.Null(parsedArgs.ErrorLogOptions);
-            Assert.False(parsedArgs.CompilationOptions.ReportSuppressedDiagnostics);
+            string[] InvalidSarifVersions = new string[] { @"C:\MyFolder\MyBinary.xml,version=1.0.0", @"C:\MyFolder\MyBinary.xml,version=2.1.0", @"C:\MyFolder\MyBinary.xml,version=42" };
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:4085]: >InvalidSarifVersions [](start = 21, length = 20)

Locals names shouldn't start with a capital letter.

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs [hunk 2]
```diff
@@ -4082,19 +4082,23 @@ public void ParseErrorLog()
...
+
+            foreach (string InvalidSarifVersion in InvalidSarifVersions)
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:4087]: >InvalidSarifVersion [](start = 28, length = 19)

Here as well

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 3)

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs (hunk 1 above)
Comment (eddynaka) [reply src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs]: Fixed. thank you!

---
In reply to: [552083212](https://github.com/dotnet/roslyn/pull/50238#discussion_r552083212) [](ancestors = 552083212)

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs (hunk 2 above)
Comment (eddynaka) [reply src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs]: Fixed. thank you!

---
In reply to: [552085375](https://github.com/dotnet/roslyn/pull/50238#discussion_r552085375) [](ancestors = 552085375)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off.

**Review by 333fred — APPROVED** [review]
LGTM (commit 4)

**Pull request merged**